src/main/java/bgu/spl/mics/application/MI6Runner.java

```diff
@@ -54,7 +54,9 @@ public static void main(String[] args) {
         e.execute(timeService);
         while (!timeService.isToTerminate()) {
```

Comment:
synchronized should wrap the while and not the opposite
(here mainly for convention sake)